### PR TITLE
Fixed branch new-argparse compilation on OS X

### PR DESCRIPTION
**Pull request by LRFLEW**

### CODE REVIEW

File: src/core/Memory.hpp [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #pragma once
 
+extern "C" {
+#include "../platform/platform.h"
+}
+
```
Comment (IntelOrca) [inline src/core/Memory.hpp]: You might not need this anymore, try rebasing.

File: src/core/String.cpp [hunk 2]
```diff
@@ -73,8 +73,7 @@ namespace String
         for (size_t i = 0; i < minSize; i++)
         {
             *dst++ = *src;
-            if (*src == '\0') break;
-            *src++;
+            if (*src++ == '\0') break;
```
Comment (IntelOrca) [inline src/core/String.cpp]: Why do this? src isn't used after breaking.

Comment (LRFLEW) [inline src/core/String.cpp]: It's less lines of code. (and the other way was giving a warning due to the dereference). If you want, I can easily revert that.

File: src/core/Memory.hpp (hunk 1 above)
Comment (LRFLEW) [inline src/core/Memory.hpp]: It doesn't look like the platform.h file itself is needed, but it's complaining about no declaration of `malloc` and such.

File: src/core/String.cpp (hunk 2 above)
Comment (IntelOrca) [inline src/core/String.cpp]: The dereference was wrong, but I am fine with keeping it like this.

File: src/core/Memory.hpp (hunk 1 above)
Comment (IntelOrca) [inline src/core/Memory.hpp]: ok, just change it to common.h instead of platform.

**Pull request merged**